### PR TITLE
Grab Scopes from Token Objects instead of Access Token

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
- [x] Remove reference to decoded Access token
  - tokenHandlerClient
  - saveDocumentStateStrategy
- [x] Unit tests work
- [x] Regression tests work

### CODE REVIEW

File: oauth-proxy/oauthHandlers/tokenHandlerStrategyClasses/saveDocumentStrategies/saveDocumentStateStrategy.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ class SaveDocumentStateStrategy {
     try {
       if (document.launch && tokens.access_token) {
         let decodedToken = jwtDecode(tokens.access_token);
-        if (decodedToken.scp.includes("launch")) {
+        if (tokens.scope && tokens.scope.includes("launch")) {
```
Comment (beaugrantham-va) [inline oauth-proxy/oauthHandlers/tokenHandlerStrategyClasses/saveDocumentStrategies/saveDocumentStateStrategy.js:61]: `decodedToken.scp` is an array while `tokens.scope` is a string.

Using `.includes` will result in different behavior here. It needs to match `launch` exactly.

File: oauth-proxy/oauthHandlers/tokenHandlerStrategyClasses/tokenHandlerClient.js [hunk 2]
```diff
@@ -68,14 +68,15 @@ class TokenHandlerClient {
...
+
+    if (tokens && tokens.scope) {
+      if (tokens.scope.includes("launch/patient")) {
         let patient = await this.getPatientInfoStrategy.createPatientInfo(
           tokens,
           decoded
         );
         responseBody["patient"] = patient;
-      } else if (decoded.scp.includes("launch") && launch) {
+      } else if (tokens.scope.includes("launch") && launch) {
```
Comment (beaugrantham-va) [inline oauth-proxy/oauthHandlers/tokenHandlerStrategyClasses/tokenHandlerClient.js:79]: `decoded.scp` is an array while `tokens.scope` is a string.

Using `.includes` will result in different behavior here. It needs to match `launch` exactly.

**Review by beaugrantham-va — CHANGES_REQUESTED** [review]

**Comment by lukemarkwordtlibertyits** [comment]
To test Regex:

```js
let arr = "launch patient/launch".split(/(\s)/);

arr.forEach(element => console.log(element));
```

https://jsfiddle.net/vh93u851/4/

**Comment by beaugrantham-va** [comment]
> To test Regex:
> 
> ```js
> let arr = "launch patient/launch".split(/(\s)/);
> 
> arr.forEach(element => console.log(element));
> ```
> 
> https://jsfiddle.net/vh93u851/4/

Using a capture group in this regular expression populates the array with unnecessary elements (size = 3).

> If separator is a regular expression with capturing parentheses, then each time separator matches, the results (including any undefined results) of the capturing parentheses are spliced into the output array.

We should drop the capture group or use a string literal (`' '`) to reduce the number of elements to search through (size = 2).

**Review by beaugrantham-va — APPROVED** [review]

**Pull request merged**